app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEIViewModel.kt

```diff
@@ -811,7 +812,15 @@ class SearchTEIViewModel(
         is FormIntent.OnFocus -> {
             val updatedItems = uiState.items.map {
                 if (it.focused && it.uid != formIntent.uid) {
-                    (it as FieldUiModelImpl).copy(focused = false)
+                    val validation = it.value?.let { value -> it.valueType?.validator?.validate(value) }
```

Comment:
Here it will be validated when it is focused right?, wouldn't be better when it is unfocused?
Also here is validating all valueTypes, in the issue where some specific, wouldn't better to filter by these valuetypes?

Reply:
This is validating the value when the field looses the focus.
I added all validations, as it was before. But i could filter it if needed.